### PR TITLE
[wasm] Load assets earlier

### LINKED ISSUE #52136: [wasm] System.Xml.Xsl.XslCompiledTransformApi tests fail on windows/helix
Similar issue to `System.IO.Compression` tests:
```
[15:59:49] fail: [FAIL] System.Xml.Tests.CTransformRegressionTest.RegressionTest1(xslInputType: URI, readerType: XmlValidatingReader, outputType: Stream, navType: XPathDocument)
[15:59:49] info: System.IO.FileNotFoundException : Could not find file '/TestFiles/TestData/XsltApiV2/Straße.xml'.
[15:59:49] info:    at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
[15:59:49] info:    at Interop.CheckIo(Error error, String path, Boolean isDirectory, Func`2 errorRewriter)
[15:59:49] info:    at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
[15:59:49] info:    at System.IO.Strategies.FileStreamHelpers.OpenHandle(String path, FileMode mode, FileAccess access, FileShare share, FileOptions options)
[15:59:49] info:    at System.IO.Strategies.Net5CompatFileStreamStrategy..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
[15:59:49] info:    at System.IO.Strategies.FileStreamHelpers.ChooseStrategy(FileStream fileStream, String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
[15:59:49] info:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
[15:59:49] info:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean useAsync)
[15:59:49] info:    at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
[15:59:49] info:    at System.IO.File.Open(String path, FileMode mode, FileAccess access, FileShare share)
[15:59:49] info:    at System.IO.File.Open(String path, FileMode mode)
[15:59:49] info:    at System.Xml.Tests.CTransformRegressionTest.RegressionTest1(XslInputType xslInputType, ReaderType readerType, OutputType outputType, NavType navType)
[15:59:49] info:    at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
[15:59:49] info: Finished:    System.Xml.Xsl.XslCompiledTransformApi.Tests.dll
[15:59:49] info: 
[15:59:49] info: === TEST EXECUTION SUMMARY ===
[15:59:49] info: Total: 1615, Errors: 0, Failed: 1, Skipped: 29, Time: 22.261575s
```

### PR DESCRIPTION
**Pull request by pavelsavara**
## Assets download

- split asset download from copying it into memory
    - do the async fetch in `Module.preInit`
    - do `mono_wasm_after_runtime_initialized` synchronously on `postRun`. It loads assets into memory. This is when `malloc` is ready. 
    - It also keeps `onRuntimeInitialized` free for user code
    - If we do something similar in Blazor, it would solve https://github.com/dotnet/aspnetcore/issues/13915
    - throttle parallel download of assets to prevent `net::ERR_INSUFFICIENT_RESOURCES`
- moved startup sequence to `configure_emscripten_startup` and documented it there
- allow `Module.config` to have assets with `buffer` instead of url
- pass config into `onConfigLoaded` and made it async
- change sample to show how emscripten callbacks are triggered
- `/p:WasmXHarnessMonoArgs="--diagnostic_tracing"` as CLI parameter for tests.

## CI changes
- move tests which fail on V8 to browser or nodeJS scenarios. 
    - Fixes https://github.com/dotnet/runtime/issues/52136
- new `browserornodejs` `<TestArchiveTestsDir>` for things which don't run on V8

### CODE REVIEW

File: src/mono/wasm/runtime/startup.ts [hunk 1]
```diff
@@ -302,6 +384,11 @@ export function bindings_lazy_init(): void {
 
 // Initializes the runtime and loads assemblies, debug information, and other files.
 export async function mono_load_runtime_and_bcl_args(config: MonoConfig | MonoConfigError | undefined): Promise<void> {
```
Comment (maraf) [inline src/mono/wasm/runtime/startup.ts:386]: Is this method still needed?

**Review by maraf — APPROVED** [review]
LGTM

File: src/mono/wasm/runtime/startup.ts (hunk 1 above)
Comment (pavelsavara) [reply src/mono/wasm/runtime/startup.ts:386]: It was on `MONO` API for some time now ... not sure about it.

**Pull request merged**